plugins/src/android/OptimoveInitProvider.java

```diff
@@ -10,30 +10,49 @@
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+
 import com.optimove.android.Optimove;
 import com.optimove.android.OptimoveConfig;
+import com.optimove.android.optimobile.OptimoveInApp;
 
 public class OptimoveInitProvider extends ContentProvider {
-    private static final String OPTIMOVE_CREDENTIALS = "optimoveCredentials";
-    private static final String OPTIMOVE_MOBILE_CREDENTIALS = "optimoveMobileCredentials";
+    private static final String KEY_OPTIMOVE_CREDENTIALS = "optimoveCredentials";
+    private static final String KEY_OPTIMOVE_MOBILE_CREDENTIALS = "optimoveMobileCredentials";
+    private static final String KEY_IN_APP_CONSENT_STRATEGY = "inAppConsentStrategy";
+
+    private static final String IN_APP_AUTO_ENROLL = "auto-enroll";
+    private static final String IN_APP_EXPLICIT_BY_USER = "explicit-by-user";
 
     @Override
     public boolean onCreate() {
         Application app = (Application) getContext().getApplicationContext();
         String packageName = app.getPackageName();
         Resources resources = app.getResources();
-        String optimoveCredentials = getStringConfigValue(packageName, resources, OPTIMOVE_CREDENTIALS);
-        String optimoveMobileCredentials = getStringConfigValue(packageName, resources, OPTIMOVE_MOBILE_CREDENTIALS);
-        if (TextUtils.isEmpty(optimoveCredentials) || TextUtils.isEmpty(optimoveMobileCredentials)) {
-            return true;
+        String optimoveCredentials = getStringConfigValue(packageName, resources,KEY_OPTIMOVE_CREDENTIALS);
+        String optimoveMobileCredentials = getStringConfigValue(packageName, resources,KEY_OPTIMOVE_MOBILE_CREDENTIALS);
+        optimoveCredentials = handleNullValues(optimoveCredentials);
+        optimoveMobileCredentials = handleNullValues(optimoveMobileCredentials);
+        String inAppConsentStrategy = getStringConfigValue(packageName, resources, KEY_IN_APP_CONSENT_STRATEGY);
+        assert (optimoveCredentials != null || optimoveMobileCredentials != null);
```

Comment:
shall we use `assert` in prod code?

https://softwareengineering.stackexchange.com/questions/137158/is-it-better-to-use-assert-or-illegalargumentexception-for-required-method-param

Reply:
ok I'll change it